slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -477,6 +487,16 @@ void SLAMServiceImpl::RunSLAM() {
 
         map_builder.map_builder_->FinishTrajectory(trajectory_id);
         map_builder.map_builder_->pose_graph()->RunFinalOptimization();
+        auto local_poses = map_builder.GetLocalSlamResultPoses();
+        if (local_poses.size() > 0) {
+            tmp_global_pose =
+                map_builder.GetGlobalPose(trajectory_id, local_poses.back());
+        }
+    }
+
+    {
+        std::lock_guard<std::mutex> lk(viam_response_mutex);
+        latest_global_pose = tmp_global_pose;
```

Comment:
[opt] setting latest_global_pose directly on line 192 otherwise latest_global_pose doesnt need to be updated

Reply:
with how offline mode currently tries to optimize at the end of the run, we need to make sure our final pose gets updated after that optimization. With out this final update our mutex protected `latest_global_pose` ends up out of date.

Reply:
when a map exists wont local_poses.size always be greater than 0 so we always count on it being updated at the end of the run? And if you are worried about the mutex we can add it around the if statement. Unless I'm misunderstanding something

Reply:
yeah thats true, i can get rid of the check at the end here. We only need it at the start of a run/map